### DISCUSSION TITLE
Copy selection coordinates into RAM

### DISCUSSION THREAD

**Discussion by Shohreh** [sumatrapdfreader/sumatrapdf#3549] [Ideas]
Hello,

I sometimes need to crop a PDF, but found no Windows application that would let me make a selection, and get its coordinates (left, bottom, right, top).

I suggest letting the user do this, and copy the coordinates into RAM, ready to be pasted in a command/script.

Cheers,

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/1541861/4a7711cd-88ee-46c1-a68b-d69e9cc20259)

**Comment by GitHubRulesOK** [comment]
hmm
 text is co-ordinate driven from lower left like maps/charts
 but they are not necessarily per line so SumatraPDF based on MuPDF uses a readout based on screen layout (0 is page Top Left)
You can use the Measure tool to read those values, but as transient they are not stored for query. hover the cursor lower left you will see its location and the width and height of the selection, but one issue is which units you chose from the 3M's
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/8e49974d-6388-432b-9c3c-9463741aaeb9)

If I recollect correctly AutoHotKey can capture the readout and with a bit of maths Nod5 was able to use that for older annotation jumping

**Comment by Shohreh** [comment]
Thanks. In the meantime, [I did find](https://graphicdesign.stackexchange.com/questions/75008/free-tool-to-view-millimeter-coordinate-under-mouse-of-a-pdf) that very useful feature.

I just need to add and then delete a redaction annotation, so need the selection's x1,y1,x2,y2 coordinates.

None of the three units (pt, mm, in) work as expected after typing those infos in my script: The area is not cropped.

```
#pt
rect = fitz.Rect(152,273,171,71) #left start, bottom start, right end, top end (rectangle is from left,bottom to right,top)
page.add_redact_annot(rect)
page.apply_redactions()
doc.save("redacted.pdf")
```

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/1541861/22aff61f-c0b1-43b4-b456-d95f40976220)

**Reply by GitHubRulesOK** [reply]
If thats fitz from PyMuPDF etc then again as far as  I know they follow MuPDF top left origin but I can be wrong, as I dont use PyMuPDF, simple test punch a hole at 100, 200,300,400 and see where it cuts. is it near left top or bottom ?

**Comment by Shohreh** [comment]
Solved: I assumed the "Selection" info was the right,top, but it's the size of the selection.

The solution is simply to 1) move the mouse to the left,bottom of the part you want to crop, 2) note the coordinates, 3) move it to the right,top, 4) note the coordinates, and crop the PDF.

Thanks!
